src/BizHawk.Client.EmuHawk/Sound/Output/DirectSoundSoundOutput.cs

```diff
@@ -90,55 +124,77 @@ public void StartSound()
 					SizeInBytes = BufferSizeBytes
 				};
 
-			_deviceBuffer = new SecondarySoundBuffer(_device, desc);
-
-			_actualWriteOffsetBytes = -1;
-			_filledBufferSizeBytes = 0;
-			_lastWriteTime = 0;
-			_lastWriteCursor = 0;
+			while (_deviceBuffer == null)
+			{
+				try
+				{
+					_deviceBuffer = new SecondarySoundBuffer(_device, desc);
+				}
+				catch (DirectSoundException)
+				{
+					System.Threading.Thread.Sleep(10);

```

Comment:
This seems very hazardous.  If, for any `DirectSoundException` reason, the buffer cannot be acquired at all, we'll endlessly loop.

Reply:
I can have it try once per frame instead, it might take a little longer to reacquire but the rest of the code should now be tolerant to being in a non-recovered situation. Or I can change it to break out of the loop after N retries, which would you prefer?

Reply:
I assume there are real world situations where the buffer will fail to create but it will succeed after Sleep(10)?  DirectSound is such a drag.

How about N retries?

Reply:
I'll have to add a check for null _deviceBuffer in StartPlaying and attempt to create it there as well as currently there is no other attempt to create the buffer other than StartSound which is called on start and window resizing. Otherwise sound would not be able to return.

If there is a persistent exception being thrown, performance is going to tank with the sleep. I'll have the N retry counter fall off to the point where if it's consistent, it'll only try once per frame. Start at 5 and drop by one each time it  doesn't succeed.